src/TheThingsNetwork.cpp

```diff
@@ -237,8 +236,10 @@ bool TheThingsNetwork::personalize() {
 bool TheThingsNetwork::provision(const byte appEui[8], const byte appKey[16]) {
   sendCommand(F("mac set appeui"), appEui, 8);
   sendCommand(F("mac set appkey"), appKey, 16);
-  sendCommand(F("mac save"));
-  return waitForOK();
+  while (!sendCommand(F("mac save"))) {
+    delay(1000);
+  }
+  return waitForOK(10000);
```

Comment:
`sendCommand()` shouldn't take long. Also, with the `while` loop, your repeating sending the command.

Just one `sendCommand()` and a long wait in `waitForOK()`